aicsimageio/readers/ome_tiff_reader.py

```diff
@@ -0,0 +1,584 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from copy import deepcopy
+import logging
+from typing import Dict, List, Tuple, Union
+import re
+import xml.etree.ElementTree as ET
+
+from fsspec.spec import AbstractFileSystem
+import numpy as np
+from ome_types import from_xml
+from ome_types.model.ome import OME
+from tifffile import TiffFile, TiffFileError, TiffTag
+import xarray as xr
+
+from .. import constants, exceptions, transforms, types
+from ..dimensions import DimensionNames, DEFAULT_DIMENSION_ORDER
+from ..utils import io_utils
+from .tiff_reader import TiffReader
+from ..types import PhysicalPixelSizes
+
+
+###############################################################################
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+)
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+# This is a known issue that could have been caused by prior versions of aicsimageio
+# due to our old OMEXML.py file.
+#
+# You can see the PR that updated this exact line here:
+# https://github.com/AllenCellModeling/aicsimageio/pull/116/commits/e3f9cde7f680edeef3ef3586a67fd8106e746167#diff-46a483e94af833f7eaa1106921191fed5e7c77f33a5c0c47a8f5a2d35ad3ba96L47
+#
+# Notably why this is invalid is that the 2012-03 schema _doesn't exist_
+#
+# Don't know how this wasn't ever caught before that PR but to ensure that we don't
+# error in reading the OME in aicsimageio>=4.0.0, we manually find and replace this
+# line in OME xml prior to creating the OME object.
+KNOWN_INVALID_OME_XSD_REFERENCES = [
+    "www.openmicroscopy.org/Schemas/ome/2013-06",
+    "www.openmicroscopy.org/Schemas/OME/2012-03",
+]
+REPLACEMENT_OME_XSD_REFERENCE = "www.openmicroscopy.org/Schemas/OME/2016-06"
+
+###############################################################################
+
+
+class OmeTiffReader(TiffReader):
+    @staticmethod
+    def _clean_ome_xml_for_known_issues(xml: str) -> str:
```

Comment:
I recommend moving this function out into a module in a `metadata/` subdirectory, so it doesn't have to clutter the reader module.

Reply:
Agree, will move.